src/pages/search/advanced_search/FileStatusSelect.tsx

```diff
@@ -0,0 +1,50 @@
+import { FormControl, InputLabel, MenuItem, Select, SelectChangeEvent } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router-dom';
+import { ResultParam } from '../../../api/searchApi';
+import { dataTestId } from '../../../utils/dataTestIds';
+
+enum FileStatus {
+  hasPublicFiles = 'hasPublicFiles',
+  noFiles = 'noFiles',
+}
+
+export const FileStatusSelect = () => {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const searchParams = new URLSearchParams(history.location.search);
+  const selectedParam = searchParams.get(ResultParam.Files) ?? '';
+
+  const handleChange = (event: SelectChangeEvent) => {
+    const newValue = event.target.value;
+
+    if (newValue !== '') {
+      searchParams.set(ResultParam.Files, newValue);
+      history.push({ search: searchParams.toString() });
+    } else {
+      searchParams.delete(ResultParam.Files);
+      searchParams.delete(ResultParam.From);
+      history.push({ search: searchParams.toString() });
+    }
```

Comment:
For å håndtere om `newValue` skulle blir null er undefined også, i tilleg til ha `history.push()` bare en plass
```suggestion
    if (newValue) {
      searchParams.set(ResultParam.Files, newValue);
    } else {
      searchParams.delete(ResultParam.Files);
      searchParams.delete(ResultParam.From);   
    }
    history.push({ search: searchParams.toString() });
```